docs/architecture/deep-dives/passkeys/operations.md

```diff
@@ -0,0 +1,148 @@
+     the credential to be used for data encryption.
+
+3. **Client-Side WebAuthn Request:** The application receives the options object from the server and
+   passes it to `navigator.credentials.create()`.
+
+4. **Client-Side Credential Creation Request**: The WebAuthn Client receives the options object,
+   validates it, searches for connected authenticators, and then sends the options object to all
+   connected authenticators to create a new FIDO2 credential.
+
+5. **User Interaction**: All connected authenticators will indicate that they have received the
+   request and are ready to create a new credential (e.g. using a blinking LED). The user will then
+   need to perform an action to confirm the creation of the new credential (e.g. by pressing a
+   button on the authenticator).
+
+6. **Authenticator Response**: Once the user confirms the creation of the new credential, the
+   authenticator generates a new public-private key pair and creates a new credential object. This
+   object includes the `credentialId`, public key, a signature, the type of authenticator used, the
+   attestation statement, and other relevant information.
+
+7. **Client-Side Response**: Once the credential is created, the client sends a request back to the
+   server. This request includes the newly created credential and all the other relevant information
+   received from the authenticator.
+
+8. **Server-Side Validation**: The server receives the request from the client and performs several
+   checks to validate the authenticity and integrity of the received data. This includes verifying
+   the challenge, verifying the signature of the credential, checking the attestation statement,
+   etc. This is usually performed by a code library.
+
+9. **Server-Side Credential Storage**: If the validation is successful, the server stores the
+   credential securely in its database, associating it with the user's account or any other relevant
+   information.
+
+10. **Registration Completion**: Finally, the server responds to the client, indicating that the
+    registration process was successful. At this point, the user has successfully registered a FIDO2
+    credential that can be used for future authentication.
+
+The private key generated during the credential creation process is always protected by the
+authenticator/passkey provider, ensuring that the user's credentials are protected even if the
+server is compromised.
+
+## Authenticating with a Passkey
+
+Authentication requests are invoked using the `navigator.credentials.get()` method.
+
+1. **Client-Side Request**: The application initiates the registration process by making a request
+   to the server. This request is typically triggered by a user action, such as entering a username,
+   or by clicking on a "Login with Passkey" button.
+
+2. **Server-Side Preparation**: The server prepares for the authentication process by generating an
```

Comment:
⛏️ Perhaps split out the (a) and (b) here into sub-bullets?

Reply:
Like this?